### ISSUE TITLE
Should all the types in ConnectionOptions be `| null`?

### ISSUE THREAD

**Issue by sgronblo** [iceddev/pg-connection-string#19]
I tried to use this with pg-promise in this way:

```
const connectionOptions = parse(connectionString)
const db = pgp({
  ...connectionOptions,
  ...myAdditionalOptions
})
```

Unfortunately this doesn't work since `pg-promise`'s TConnectionParameters has it's properties defined as `host?: string` which translates to `host: string | undefined` so they are incompatible. I had a brief look at the source code for `pg-connection-string` and I am not sure all the properties are actually `null`, in fact there is only a single explicit fallback to `null` in the whole source file.

**Comment by hjr3** [comment]
Being worked on here https://github.com/iceddev/pg-connection-string/pull/18

**Comment by hjr3** [comment]
I believe this is now fixed in #18.
